File: 0330/solution.ts
function minPatches(nums: number[], n: number): number {
  let range = 0;
  let res = 0;
  let idx = 0;
  while (range < n) {
    if (idx == nums.length || range < nums[idx] - 1) {
      res++;
      range += range + 1;
    } else {
      range += nums[idx++];
    }
  }
  return res;
}

File: 0330/solution.py
class Solution:
    def minPatches(self, nums: list[int], n: int) -> int:
        res = 0
        num = 0
        i = 0
        while num + 1 <= n:
            if i < len(nums) and nums[i] <= num + 1:
                num += nums[i]
                i += 1
            else:
                res += 1
                num += num + 1
        return res
